ex4/ex4.c

```diff
@@ -1,15 +1,79 @@
 // Write a program that calls `fork()` and then calls some form of `exec()`
-// to run the program `/bin/ls`. Try a few variants of `exec()`, such as 
-// `execl()`, `execle()`, `execv()`, and others. Why do you think there 
+// to run the program `/bin/ls`. Try a few variants of `exec()`, such as
+// `execl()`, `execle()`, `execv()`, and others. Why do you think there
 // are so many variants of the same basic call?
 
+//
+
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <string.h>
 
-int main(int argc, char* argv[])
+int main(int argc, char *argv[])
 {
-    // Your code here    
+	// Your code here
+	printf("pid before fork(): %d\n", (int)getpid());
+	int rc = fork();
+	// --- child process start here ---
+
+	if (rc < 0)
+	{
+		fprintf(stderr, "fork blew up, try again! \n");
+		exit(1);
+	}
+	else if (rc == 0)
+	{
+		printf("child pid: %d\n", (int)getpid());
+		char *myargs[3];
+
+		myargs[0] = strdup("/bin/ls");
+		myargs[1] = strdup("./");
+		myargs[2] = NULL;
+
+		// execvp(myargs[0], myargs);
+		execl(myargs[0], myargs[1], myargs[2]);
```

Comment:
When compiling this code, this line throws the warning `missing sentinel in function call [-Wformat=]` - It would definitely be worthwhile to look up why that occurs, and how to prevent it from happening.